src/write.c

```diff
@@ -2054,12 +2428,14 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
 
         for (uint32_t itemIndex = 0; itemIndex < encoder->data->items.count; ++itemIndex) {
             avifEncoderItem * item = &encoder->data->items.item[itemIndex];
-            const avifBool isGrid = (item->gridCols > 0); // Grids store their payload in metadataPayload, so use this to distinguish grid payloads from XMP/Exif
+            // Grids and tmap store their payloads in metadataPayload, so use this to distinguish grid payloads from XMP/Exif
+            const avifBool isMetadata = (!memcmp(item->type, "mime", 4) || !memcmp(item->type, "Exif", 4)) &&
+                                        (item->metadataPayload.size > 0);
```

Comment:
I think `&& (item->metadataPayload.size > 0)` is unnecessary here, and the comment can be removed in my opinion